CONTRIBUTING.md

```diff
@@ -1,50 +1,75 @@
+:balloon: Thanks for your help improving the project!
 
 ## Getting Help ##
 
-If you have a question about linkerd-tcp or have encountered problems
-using linkerd-tcp, you may [file an issue][issue] or join us on
-[Slack][slack].
+If you have a question about linkerd-tcp or have encountered problems using it, you may
+[file an issue][issue] or join us on [Slack][slack].
 
-## Submitting Changes ##
+## Submitting a Pull Request ##
 
-If you'd like to make a change to linkerd-tcp, you should do the
-following:
+Do you have an improvement?
 
 1. Submit an [issue][issue] describing your proposed change.
-2. We will respond to your issue promptly.
-3. If your proposed change is accepted, and you haven't already done
-so, sign [Buoyant's Contributor License Agreement][cla].  Before we
-can accept your patches, our lawyers would like to be assured that:
-    - The code you're contributing is yours, and you have the right to
-    license it.
-    - You're granting us a license to distribute said code under the
-    terms of this agreement.
-4. Fork the linkerd-tcp repo, develop and test your code
-changes. See the project's [README](README.md) for further information
-about working in this repository.
-5. Submit a pull request against linkerd-tcp's `master` branch.
-6. Your branch may be merged once all configured checks pass,
-including:
+2. We will try to respond to your issue promptly.
+3. If your proposed change is accepted, and you haven't already done so, sign our [Contributor License Agreement][cla].
+4. Fork this repo, develop and test your code changes. See the project's [README](README.md) for further information about working in this repository.
+5. Submit a pull request against this repo's `master` branch.
+6. Your branch may be merged once all configured checks pass, including:
     - 2 code review approvals, at least 1 of which is from a buoyant member
     - The branch has passed tests in CI.
 
-### Git history ###
+## Committing
 
-Prior to submitting a pull request, please revise the commit history
-of your branch such that each commit is self-explanatory, even without
-the context of the pull request, and that it compiles & tests
-cleanly.
+We prefer squash or rebase commits so that all changes from a branch are committed to
+master a single commit.
 
-Typically, a pull request consists of a single commit.  It may
-occasionally be preferable to submit a change with several commits,
-but each should be a complete change needed to complete a larger
-feature.  Each commit should be documented appropriately.
+### Commit messages
 
-Thank you for getting involved!
-:heart: Team Buoyant
+Finalized commit messages should be in the following format:
+
+```
+Subject
+
+Problem
+
+Solution
+
+Validation
+```
+
+#### Subject
+
+- one line, <= 50 characters
+- describe what is done; not the result
+- use the active voice
+- capitalize properly
+- do not end in a period — this is a title/subject
+- reference the pull request by number
```

Comment:
Should this be "reference the github issue by number"? The PR number won't be known at the time they're creating the commit message.